flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/common/TieredStorageUtils.java

```diff
@@ -29,6 +34,18 @@ public class TieredStorageUtils {
         '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'
     };
 
+    public static List<TierFactory> getTierFactoriesFromConfiguration() {
+        TieredStorageConfiguration tieredStorageConfiguration =
+                TieredStorageConfiguration.builder()
+                        .setTierTypes(TieredStorageConfiguration.memoryDiskTierTypes())
+                        .build();
```

Comment:
This is weird.
1. We get an array of types out of `TieredStorageConfiguration`, then set it back to `TieredStorageConfiguration#Builder`.
2. The method `xxxFromConfiguration` does not accept a configuration as input. Instead, it creates a configuration internally.

I assume 2 is a temporal status, which should be commented. In addition, how is `TieredStorageConfiguration` supposed to be generated in the final state?

Reply:
Removed the `getTieredFactories` in `TieredStorageUtis`. In addition, the `TieredStorageUtis` is useless now, so I also removed it. Now we can get the factories with `TieredStorageConfiguration#getTierFactories`.

Removed the `setTierTypes`, because in the first version, we need not set the tier types from outside of the configuration.